lib/strings/ansi.rb

```diff
@@ -12,7 +12,13 @@ module ANSI
     RESET = "\e[0m"
 
     # The regex to match ANSI codes
-    ANSI_MATCHER = '(\[)?\033(\[)?[;?\d]*[\dA-Za-z]([\];])?'
+    ANSI_MATCHER = %r{
+      (?>(\[)?\033(\[)?[;?\d]*[\dA-Za-z]([\];])?)
+      |
+      # see https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda#the-escape-sequence
+      # hyperlink opening sequence (closing sequence is already covered by previous part)
+      (?>\033\]8;[^;]*;.*?(\033\\|\07))
+    }x
```

Comment:
Do you think it would possible to make this into a single regexp and merge the opening hyperlink sequence? Given that the closing sequence is already covered. My preference would be also to remove the comments eventually leaving the `hyperlink opening sequence` if the union cannot be easily merged.

Reply:
When quickly looking through the regex, apart from some level of duplication, I was also thinking of the performance. From my limited experience, regexp unions are akin to doing another loop/match process. This is probably a case of premature optimization and I agree with you that we should stick with readability first. I will add performance tests anyhow to ensure that the regex in future stays performant or can be improved. Before I do so, I'd like to add also tests for all the VT100 escape codes like cursor movement to ensure that they're all handled.

I'm convinced, I like your version, let's stay with it minus the comments.